File: 4. Largest Palindrome Product/solution.ts
function isPalindrome(n: number): boolean {
    return n.toString() == n.toString().split('').reverse().join('')
}

function largestPalindrome(): number | null {
    for (let num = 999 * 999; num >= 100 * 100; num--) {
        if (isPalindrome(num)) {
            for (let i = 999; i >= 99; i--) {
                if (num % i === 0 && 100 <= num / i && num / i <= 999) {
                    return num;
                }
            }
        }
    }
    return null;
}

console.log(largestPalindrome())

File: 4. Largest Palindrome Product/solution.java
public class solution {

    public static boolean isPalindrome(int n) {
        String strN = Integer.toString(n);
        return strN.equals(new StringBuilder(strN).reverse().toString());
    }

    public static int largestPalindrome() {
        for (int num = 999 * 999; num >= 100 * 100; num--) {
            if (isPalindrome(num)) {
                for (int i = 999; i >= 100; i--) {
                    if (num % i == 0 && 100 <= num / i && num / i <= 999) {
                        return num;
                    }
                }
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println(largestPalindrome());
    }
}